### PR TITLE
feature/folder-size

### PR DESCRIPTION
**Pull request by sraver**
From the front we need to know the folder size in order to take some decisions.

I just implemented a new endpoint to compute the size and send it back to the client.

### CODE REVIEW

**Comment by sraver** [comment]
> The implementation is good but where are you planning to use it?

The front is not going to allow sharing a folder, and when the user asks for that it will need to ask for the size so it can take some decisions.

File: src/app/services/folder.js [hunk 1]
```diff
@@ -562,6 +562,48 @@ module.exports = (Model, App) => {
...
+        [sequelize.fn('sum', sequelize.col('size')), 'total']
+      ],
+      raw: true,
+      where: {
+        folderId: { [Op.eq]: folderId },
+        userId: { [Op.eq]: userId },
+      },
+    });
+
+    return result[0].total;
```
Comment (sg-gs) [inline src/app/services/folder.js:604]: Does `result` always have one element at least? Does this first element always contain a `total` attribute? Just for ensuring that this function is fault-tolerant

File: src/app/routes/storage.js [hunk 2]
```diff
@@ -25,6 +25,20 @@ module.exports = (Router, Service, App) => {
   const sharedAdapter = sharedMiddlewareBuilder.build(Service);
   const teamsAdapter = teamsMiddlewareBuilder.build(Service);
 
+  Router.get('/storage/folder/size/:id', passportAuth, sharedAdapter, (req, res) => {
+    const { params, behalfUser } = req;
+    const folderId = params.id;
+    Service.Folder.getSize(behalfUser, folderId)
+      .then((size) => {
```
Comment (sg-gs) [inline src/app/routes/storage.js:32]: `async-await` could be used here without catching any error as we have a catch-all middleware, thus making code more readable

**Review by sg-gs — CHANGES_REQUESTED** [review]

File: src/app/services/folder.js (hunk 1 above)
Comment (sraver) [reply src/app/services/folder.js]: yes it has always that structure

**Review by sg-gs — APPROVED** [review]

**Pull request merged**